src/server/pfs/s3/meta.go

```diff
@@ -0,0 +1,61 @@
+package s3
+
+import (
+	"bytes"
+	"strings"
+	"encoding/json"
+	"fmt"
+
+	"github.com/pachyderm/pachyderm/src/client"
+)
+
+// ObjectMeta is JSON-serializable metadata about an object
+type ObjectMeta struct {
+	MD5 string `json:"md5"`
+}
+
+func isMeta(file string) bool {
+	return strings.HasSuffix(file, ".s3g.json")
+}
+
+func getMeta(client *client.APIClient, repo, branch, file string) (*ObjectMeta, error) {
+	metaReader, err := client.GetFileReader(repo, branch, fmt.Sprintf("%s.s3g.json", file), 0, 0)
+	if err != nil {
+		if !fileNotFoundMatcher.MatchString(err.Error()) {
+			return nil, err
+		}
+		return nil, nil
+	}
+
+	meta := new(ObjectMeta)
+	if err = json.NewDecoder(metaReader).Decode(&meta); err != nil {
+		if !fileNotFoundMatcher.MatchString(err.Error()) {
+			return nil, err
+		}
+		return nil, nil
+	}
+
+	return meta, nil
+}
+
+func putMeta(client client.PutFileClient, repo, branch, file string, meta *ObjectMeta) error {
+	metaBytes, err := json.Marshal(meta)
+	if err != nil {
+		panic(err)
```

Comment:
This should never error, but we probably don't want to panic here and risk crashing the entire server when it's in the middle of something.

Reply:
Go's http server recovers from panics; I think this will just kill the goroutine servicing this specific client

Reply:
Confirmed that is the case. Do you still want me to change this?

Reply:
Looks like this is no longer an issue because we nixed the meta file. I think since the http handler catches it, in theory there's no strong reason we can't use a panic here. I have a general fear of panics which is why we generally don't use them except in cases where we truly do want to crash the whole process such as when the networking stack isn't even letting us connect to ourselves. There is a small performance penalty associated with panicking, but I don't think that's a compelling reason to never use it either. These are just some general thoughts though, in this case it seems to be moot since the code has been removed for other reasons.